drivers/src/hand_off.rs

```diff
@@ -427,6 +427,26 @@ impl FirmwareHandoffTable {
         None
     }
 
+    pub fn try_into_ref() -> Option<&'static FirmwareHandoffTable> {
```

Comment:
It is not sound for this function to be safe, as the aliasing rules for the returned reference will be violated as soon as somebody calls FirmwareHandoffTable::save().

This function needs to be in an actual driver that returns references with the same lifetime as the driver's &self, and the save() method needs to take the driver's (&mut self). That way, the type system will prevent aliasing violations.

Reply:
See #690 for what I'm thinking.

Reply:
Sounds good. I'm closing this PR in favor of #690